File: jiwon/053 문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    vector<pair<string, string>> aa;
    string ss;
    
    sort(strings.begin(), strings.end());
    
    for(int i=0;i<strings.size();i++){
        ss = strings[i].at(n);
        
        aa.push_back(pair<string, string>(ss, strings[i]));
    }
    
    sort(aa.begin(), aa.end());
    
    for(int i=0;i<strings.size();i++){
        answer.push_back(aa[i].second);
    }
    
    return answer;
}

File: lesunny/053_문자열내마음대로정렬하기.java
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public String[] solution(String[] strings, int n) {
        String[] answer = new String[strings.length];
        ArrayList<String> alStrings = new ArrayList<>();
        
        //input
        for(int i=0; i<strings.length; i++){
            alStrings.add(strings[i].substring(n,n+1) + strings[i]);
        }
        System.out.println(alStrings);
        
        //ascending order
        Collections.sort(alStrings);
        
        int idx = 0;
        for(String string : alStrings){
            answer[idx++] = string.substring(1, string.length());
        }
        
        return answer;
    }
}
